### PR TITLE
added schemas

### PR DESCRIPTION
**Pull request by MagdalenaBialik**

### CODE REVIEW

File: api/dynamodb/dynamodb.py [hunk 1]
```diff
@@ -8,37 +9,43 @@ def __init__(self, dynamodb_table, settings: Settings):
...
             Item={
-                "PK": city.title(),
-                "SK": attraction.title(),
+                "PK": attraction.city.title(),
+                "SK": attraction.name.title(),
             },
         )
 
-    def get_attraction(self, city, attraction):
+    def attraction_from_item(self, item: dict) -> Attraction:
```
Comment (MagdalenaBialik) [inline api/dynamodb/dynamodb.py]: `attraction_from_item -> _attraction_from_item 
`
methods that are not meant to be used outside of the class should be prefixed with _ 
https://www.datacamp.com/tutorial/role-underscore-python
This doesn't change much - just good practice

File: api/dynamodb/dynamodb.py [hunk 2]
```diff
@@ -8,37 +9,43 @@ def __init__(self, dynamodb_table, settings: Settings):
...
+        return Attraction(city=item["PK"], name=item["SK"])
+
+    def key_from_attraction(self, attraction: Attraction) -> dict:
```
Comment (MagdalenaBialik) [inline api/dynamodb/dynamodb.py]: same as above
`key_from_attraction -> _key_from_attraction`

File: api/dynamodb/dynamodb.py [hunk 3]
```diff
@@ -8,37 +9,43 @@ def __init__(self, dynamodb_table, settings: Settings):
         self.dynamodb_table = dynamodb_table
         self.settings = settings
 
-    def put_attraction(self, city: str, attraction: str):
+    def put_attraction(self, attraction: Attraction):
         self.dynamodb_table.put_item(
             Item={
-                "PK": city.title(),
-                "SK": attraction.title(),
+                "PK": attraction.city.title(),
```
Comment (MagdalenaBialik) [inline api/dynamodb/dynamodb.py]: use key_from_attraction method here

File: api/dynamodb/dynamodb.py [hunk 4]
```diff
@@ -8,37 +9,43 @@ def __init__(self, dynamodb_table, settings: Settings):
...
+                "SK": attraction.name.title(),
             },
         )
 
-    def get_attraction(self, city, attraction):
+    def attraction_from_item(self, item: dict) -> Attraction:
+        return Attraction(city=item["PK"], name=item["SK"])
+
+    def key_from_attraction(self, attraction: Attraction) -> dict:
+        key = {"PK": attraction.city, "SK": attraction.name}
```
Comment (MagdalenaBialik) [inline api/dynamodb/dynamodb.py]: you can put .title() str methods here

File: api/main.py [hunk 5]
```diff
@@ -28,32 +29,32 @@ def variable():
...
+    response = dynamodb_dao.get_attraction(attraction)
     if response is None:
         raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail="Item not found")
 
     return response
 
 
 @app.get("/get_attraction_by_city")
-def get_attraction_by_city(city):
+def get_attraction_by_city(city) -> Attraction:
```
Comment (MagdalenaBialik) [inline api/main.py]: add type hint to city

File: api/dynamodb/dynamodb.py [hunk 6]
```diff
@@ -8,37 +9,43 @@ def __init__(self, dynamodb_table, settings: Settings):
...
+    def get_attraction(self, attraction: Attraction) -> Attraction:
         response = self.dynamodb_table.get_item(
-            Key={
-                "PK": city,
-                "SK": attraction,
-            },
+            Key=self.key_from_attraction(attraction),
         )
-        return response.get("Item", None)
+        item = response.get("Item", None)
```
Comment (MagdalenaBialik) [inline api/dynamodb/dynamodb.py:29]: what if that attraction does not exist? will the code work?

File: api/exceptions.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+class NotFoundError(Exception):
+    def __init__(self, error):
```
Comment (FidelusAleksander) [inline api/exceptions.py]: https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python

remove __init__

**Pull request merged**